---
title: "mwas test"
author: "Ciao"
output: html_notebook
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(ggplot2)
library(ggpubr)
library(ggbiplot) # devtools::install_github("vqv/ggbiplot")
library(cowplot)
library(reshape2)
library(RColorBrewer)
library(plyr)
library(dplyr)
library(vegan)
library(ape)
library(pheatmap)
library(readxl)
if (!require(devtools)) install.packages("devtools")
if (!require(ggvenn)) devtools::install_github("yanlinlin82/ggvenn")
library(ggvenn)
library(plotly)



```

# 0.Metadata

```{r}
metadata.df <- read_excel("./TS21-merged-metadata.xlsx","subset",
    col_types=c(
      "numeric","text","numeric","numeric","numeric",
      "text","text","date","text","text","text","text","numeric","numeric",
      "text"),
    na="NA")

metadata.df$pushcore <- sub("(|[0-3])[0-9]\\-\\d+","", metadata.df$fmtID,perl=T)
metadata.df$fmtID <- gsub("-",".",metadata.df$fmtID,fixed = T)
metadata.df$fmtID <- gsub("[()]",".",metadata.df$fmtID,perl=T)
metadata.df$Depth <- - metadata.df$Depth
metadata.df$Collect_date <- as.POSIXlt(metadata.df$Collect_date,
                                       formats = "%Y-%m-%d")
metadata.df$day <- as.numeric(difftime(metadata.df$Collect_date,"2021-08-21 8:0:0",units="days"))

summary(metadata.df)
```

## No.dive
```{r}
fig1 <- plot_ly(metadata.df, x=~Longitude, y=~Latitude, z= ~ Depth, color= ~No.dive)

fig1
```
```{r}
repr.dive.df <- ddply(
  unique(metadata.df[,c("No.dive","Location","Longitude","Latitude")]),
  c("No.dive","Location"),summarize,
  Longitude=mean(Longitude), Latitude=mean(Latitude))

#ggplot(repr.dive.df%>%filter(grepl("Mar",Location)),
ggplot(repr.dive.df,
       aes(x=Longitude,y=Latitude)) + 
  geom_point(aes(color=No.dive)) +
  geom_text(aes(label=No.dive),hjust=1.1) + theme_bw()

ggplot(repr.dive.df%>%filter(grepl("Mar",Location)),
       aes(x=Longitude,y=Latitude)) + 
  geom_point(aes(color=No.dive)) +
  geom_text(aes(label=No.dive),hjust=1.1) + theme_bw()
```


## Collector
```{r}
fig2 <- plot_ly(metadata.df, x=~Longitude, y=~Latitude, z= ~ Depth, color= ~Collector)

fig2
```

## Collect_loc
```{r}
metadata.df$Collect_loc.s <- substr(metadata.df$Collect_loc,1,1)
fig4 <- plot_ly(metadata.df, x=~Longitude, y=~Latitude, z= ~ Depth, 
                color= ~Collect_loc.s)

fig4
```
## Location
```{r}
metadata.df$Collect_loc.s <- substr(metadata.df$Location,1,1)
fig5 <- plot_ly(metadata.df, x=~Longitude, y=~Latitude, z= ~ Depth, 
                color= ~Location)
fig5
```
# 1.MAG fpkm profile

```{r}
p.prf <- read.table(gzfile("profiles/MEER1194.d95.fpkm.MAG.profile.gz"),
                    header=T,row.names = 1)

p.shannon <- data.frame(
  MAG.shannon=diversity(p.prf,"shannon",MARGIN=2),
  MAG.simpson=diversity(p.prf,"simpson",MARGIN=2),
  MAG.invsimpson=diversity(p.prf,"invsimpson",MARGIN=2)
)
p.shannon$fmtID <- rownames(p.shannon)

mdf <-merge(metadata.df,p.shannon,by="fmtID")
```

## Alpha diversity
```{r}
ggplot(mdf,aes(x=MAG.shannon,color=Location)) + geom_density() + theme_bw()
ggplot(mdf,aes(x=MAG.simpson,color=Location)) + geom_density() + theme_bw()
ggplot(mdf,aes(x=MAG.invsimpson,color=Location)) + geom_density() + theme_bw()
```
总体分布上来看，大部分样本的shannon index可以达到4以上。雅浦海沟整体丰富度最高，马沟东部次之。
菲律宾海盆样本的多样性相对较低。

```{r}
ggplot(mdf,aes(x=MAG.shannon,y=Depth,shape=Location,color=No.dive)) + geom_point()
ggplot(mdf,aes(x=MAG.simpson,y=Depth,shape=Location,color=No.dive)) + geom_point()
ggplot(mdf,aes(x=MAG.invsimpson,y=Depth,shape=Location,color=No.dive)) + geom_point()
```
Alpha diversity 与下潜深度的关系。部分低多样性样本点在各个采样区域(Location)都有发现。


```{r}
ggplot(mdf,aes(x=MAG.shannon,y=LayerDepth,color=Collector)) + stat_density_2d() + geom_point(aes(shape=Location))
ggplot(mdf,aes(x=MAG.simpson,y=LayerDepth,color=Collector)) + stat_density_2d() + geom_point(aes(shape=Location))
ggplot(mdf,aes(x=MAG.invsimpson,y=LayerDepth,color=Collector)) + stat_density_2d() + geom_point(aes(shape=Location))
```

Alpha diversity 与下潜深度的关系。部分低多样性样本点可能与特定采样人员参与的特定潜次有关。
- 低多样性样本点多来自菲律宾海盆样本；
- 低多样性样本点多集中在10cm以下的pushcore分层中。

```{r}
ggplot(mdf,aes(x=Collector,y=MAG.shannon,fill=Collector)) + geom_violin(alpha=.5) + geom_boxplot(width=.1) + 
  theme_bw() + facet_grid(.~Location,space="free",scale="free")
```

Alpha diversity 在采样人员分组之间的比较。低多样性样本点呈现两种情况：  
- 在菲律宾海盆地区的低样本点未超出整体样本分布区间(95% CI)，具有一定普遍性；
- 在马沟内，样本整体多样性高；低丰度样本多为离群点(超出95% CI)，可能是异常状态。

* 95% CI: value < 1.58 * IQR / sqrt(n)

```{r,warning=FALSE}
ggplot(mdf,aes(x=Collector,y=MAG.shannon,fill=Collector)) + geom_violin() + geom_jitter() +
  theme_bw() + facet_wrap(~Location+No.dive,ncol=8,scale="free_x")
```

Alpha diversity 在潜次之间的比较。低多样性样本集中出现于特定潜次。其中FDZ072（Z.WS） & FDZ076(Z.WJ)出现频率最高。


```{r}
ggplot(mdf%>%filter(No.dive%in%c("FDZ072","FDZ069","FDZ076")),
       aes(x=interaction(Depth,Characteristics),y=LayerDepth,color=MAG.shannon)) + 
  geom_point(aes(shape=No.dive)) +
  scale_color_distiller(palette = "RdBu") + facet_grid(.~No.dive,scale="free",space="free") +
  theme_bw() + theme(axis.text.x = element_text(angle=-90,hjust=0))
```

低多样性样本集中出现于特定pushcore的深层位置中。



## PCA

```{r}
library(cluster)

gdf <- t(p.prf)
gdf1 <- gdf[,which(colSums(gdf)>20)]
g.pam <- pam(gdf1, k=3)
plot(g.pam)
```


```{r}
library(ggfortify)

p.prf1 <- as.data.frame(p.prf)
p.prf$taxid <- rownames(p.prf1)

p1 <- autoplot(g.pam,data=as.data.frame(gdf),alpha=.3,
               loadings=T,loadings.label=T,frame=T,frame.type="norm",
               loadings.colour="black")
p1
```



```{r}
loadings <- layer_data(p1,3)
loadings$v <- loadings$x^2+loadings$y^2
loading <- loadings[rev(order(loadings$v)),]
head(loading,10)
```


## taxonoming
```{r}
binmap.tsv <- read.table("drep_all/drep_all_95.binmap.tsv",
                         col.names = c("bin","cluster"))
drep_all_95.Cdb.csv <- read.csv("drep_all/drep_all_95.Cdb.csv")
drep_all_95.Cdb.csv$bin <- gsub(".fa","",gsub(".bin","",drep_all_95.Cdb.csv$genome))
```

```{r}
gtdbtk.ar53 <- read.table(gzfile("drep_all/drep_all_99.gtdbtk.ar53.summary.tsv.gz"),
                            sep="\t",header = T)
gtdbtk.ar53$bin <- gsub(".bin","",gtdbtk.ar53$user_genome)

bin.gtdbtk.ar53 <- merge(drep_all_95.Cdb.csv[,c("secondary_cluster","bin")],
                           gtdbtk.ar53[,c("user_genome","classification","bin")],by='bin')

bin.gtdbtk.ar53%>%filter(secondary_cluster%in%loading$label[1:6])
```

按loading向量长度排序，前6名、第10名MAG均为古菌。

注释信息简单汇总如下：
2455_2:   d__Archaea;p__Thermoproteota;c__Nitrososphaeria;o__Nitrososphaerales;
          f__Nitrosopumilaceae;g__DRGT01;s__DRGT01 sp016838795
2626_4:   d__Archaea;p__Nanoarchaeota;c__Nanoarchaeia;o__Pacearchaeales;
          f__GW2011-AR1;g__PSPM01;s__
2537_0:   d__Archaea;p__Nanoarchaeota;c__Nanoarchaeia;o__SCGC-AAA011-G17;
          f__JACPIM01;g__;s__
2466_1:   d__Archaea;p__Thermoproteota;c__Nitrososphaeria;o__Nitrososphaerales;
          f__Nitrosopumilaceae;g__Nitrosopumilus;s__Nitrosopumilus sp013390905
2455_1:   d__Archaea;p__Thermoproteota;c__Nitrososphaeria;o__Nitrososphaerales;
          f__Nitrosopumilaceae;g__DRGT01;s__
2467_1:   d__Archaea;p__Thermoproteota;c__Nitrososphaeria;o__Nitrososphaerales;
          f__Nitrosopumilaceae;
2465_1:   d__Archaea;p__Thermoproteota;c__Nitrososphaeria;o__Nitrososphaerales;
          f__Nitrosopumilaceae;g__Nitrosopumilus;s__

其中主要成员来自*氨氧化古菌*Nitrososphaeria、*亚硝化侏儒菌科*(Nitrosopumilaceae)和*纳古菌门*（Nanoarchaeota）。

综上，PCA聚类主要分为三种类型：
- c1, 由氨氧化古菌（2455_2）主导,具体为DRGT01 sp016838795
- c2, 由纳古菌（2626_4，2537_0）主导，但对应科不同，具体种未知。
- c3, 由氨氧化古菌（2455_2）主导,具体为Nitrosopumilus sp013390905


```{r}
gtdbtk.bac120 <- read.table(gzfile("drep_all/drep_all_99.gtdbtk.bac120.summary.tsv.gz"),sep="\t",header = T)
gtdbtk.bac120$bin <- gsub(".bin","",gtdbtk.bac120$user_genome)

bin.gtdbtk.bac120 <- merge(drep_all_95.Cdb.csv[,c("secondary_cluster","bin")],
                           gtdbtk.bac120[,c("user_genome","classification","bin")],by='bin')

bin.gtdbtk.bac120%>%filter(secondary_cluster%in%loading$label[1:6])
bin.gtdbtk.bac120%>%filter(secondary_cluster%in%loading$label[7:9])
```
191_1:    d__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__UBA4486;f__UBA4486;g__SMWN01;s__SMWN01 sp004356415
1471_1:   d__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__;f__;g__;s__
2340_0:   d__Bacteria;p__Zixibacteria;c__MSB-5A5;o__GN15;f__FEB-12;g__B120-G9;s__

191_1:    变形菌门；γ-变形菌纲
1471_1:   变形菌门；α-变形菌纲
2340_0:   河床菌门；

### Top10 distribution
```{r}
top10.taxon <- loading[1:10,c("label","x","y")]
top10.taxon$domain <- c(rep("Archaea",6),rep("Bacteria",3),"Archaea")
top10.taxon$class <- c("Nitrososphaeria","Nanoarchaeia","Nanoarchaeia",
                       rep("Nitrososphaeria",3),"Gammaproteobacteria",
                       "Alphaproteobacteria","Zixibacteria;MSB-5A5","Nitrososphaeria")
top10.taxon$contributes <- seq(1,10)
top10.taxon$taxon <- factor(paste0("top",top10.taxon$contributes,":",top10.taxon$class,":",top10.taxon$MAGID),
                            levels=paste0("top",top10.taxon$contributes,":",top10.taxon$class,":",top10.taxon$MAGID))


top10.prf.mdf <- melt(as.matrix(p.prf[which(rownames(p.prf)%in%top10.taxon$label),]))
top10.prf.mdf$value <- as.numeric(top10.prf.mdf$value)
pca.cluster <- as.data.frame(g.pam$clustering)
pca.cluster$fmtID <- rownames(pca.cluster)
colnames(pca.cluster)[1] <- "PCA.cluster"

colnames(top10.prf.mdf) <- c("MAGID","fmtID","FPKM")

top10.mdf <- merge(
  merge(top10.prf.mdf,top10.taxon,by.x="MAGID",by.y="label"),
  merge(mdf,pca.cluster,by="fmtID"),by="fmtID")
```

```{r}
ggplot(top10.mdf,aes(x=FPKM,color=taxon)) + 
  geom_density() + scale_x_log10()
```
```{r}
ggplot(top10.mdf,aes(x=PCA.cluster,y=FPKM,fill=factor(PCA.cluster))) + 
  geom_violin() + geom_boxplot(width=.2) + scale_y_log10() + facet_wrap(.~taxon,scale="free",ncol=5) +
  theme_bw() + scale_fill_brewer(palette="Set1") +
  stat_compare_means(label = "p.signif")
```

```{r}
table(top10.mdf$PCA.cluster,top10.mdf$Location)/10

ggplot(top10.mdf,aes(x=PCA.cluster,y=..count../10)) + geom_bar(aes(fill=factor(PCA.cluster))) + 
  facet_grid(.~Location) + theme_bw() + scale_fill_brewer(palette="Set1")
```

聚类分布有以下特征：
- 菲律宾海盆南部几乎以C1为主；
- 菲律宾海盆南北部以C1为主，少量C2，几乎无C3；
- 马沟东部以C1为主，C3次之，极少C2；
- 雅浦海沟几乎均为C3类型

此外，C3在马沟区域比例明显高于菲律宾海盆地区，提示C3可能是马沟特有的群落结构。
雅浦海沟完全没有观察到C2类型的样本，可能跟该地区的生境有所关联。


```{r}
ggplot(top10.mdf%>%filter(contributes<5),aes(x=contributes,y=FPKM,fill=taxon)) + geom_boxplot() + 
  facet_grid(.~Location) + theme_bw() + scale_y_log10() + stat_compare_means(label = "p.signif")
```





# 2.Fungal MAG fpkm profile

```{r}
f.prf <- read.table(gzfile("profiles/MEER1194.fungi.fpkm.profile.gz"),
                    header=T,row.names = 1)
f.stat <- read.table(gzfile("profiles/MEER1194.fungi.fpkm.row.stat.gz"),
                    header=T,row.names = 1)

f.stat$contig <- rownames(f.stat)
f.prf$contig <- rownames(f.prf)

f.mdf <- merge(
  melt(f.prf,id.vars = "contig",variable.name = "sample",value.name = "FPKM"),
  f.stat,by="contig")


f.mdf$MAG <- ifelse(f.mdf$contig=="deepsea_asco_mitochondria@k141_44389","TheFungi.Mitochondrion","TheFungi.MAG")

f.MAG.mdf <-ddply(f.mdf,c("sample","MAG"),summarise,FPKM=sum(FPKM*length)/sum(length))

f.MAG.prf <- dcast(f.MAG.mdf,MAG~sample,value.var = "FPKM")

f.MAG.mdf2 <- dcast(f.MAG.mdf,sample~MAG,value.var = "FPKM")
f.MAG.mdf2$ratio <- f.MAG.mdf2$TheFungi.Mitochondrion/f.MAG.mdf2$TheFungi.MAG
f.MAG.mdf2 <- merge(metadata.df,f.MAG.mdf2,by.x="fmtID",by.y="sample")
```

## correlation of mito&genome
```{r}
ggplot(f.MAG.mdf2,aes(x=TheFungi.MAG,y=TheFungi.Mitochondrion)) + 
  geom_point() + theme_bw()

ggplot(f.MAG.mdf2,aes(x=ratio)) + 
  geom_density() + theme_bw()
```

```{r}
f.MAG.mdf2$MMRatioGroup <- ifelse(
  f.MAG.mdf2$ratio>100,
  ifelse(f.MAG.mdf2$ratio>230,"round270","round170"),"round040")
```
```{r}
ggplot(f.MAG.mdf2,aes(x=TheFungi.MAG,y=TheFungi.Mitochondrion)) + 
  geom_point(aes(color=MMRatioGroup)) + theme_bw()

ggplot(f.MAG.mdf2,aes(x=TheFungi.MAG,y=TheFungi.Mitochondrion)) + 
  #geom_abline(slope = c(40,170,270),linetype=2,alpha=.3) +
  stat_smooth(aes(group=MMRatioGroup),method = "lm", se = TRUE) +
  stat_cor(label.x=30,method="spearman",aes(label.y=MMRatioGroup,group=MMRatioGroup)) +
  geom_point(aes(color=MMRatioGroup)) + theme_bw() +
  facet_wrap(~Location)
```

```{r}
ggplot(f.MAG.mdf2,aes(x=ratio,fill=MMRatioGroup)) + 
  geom_histogram() + theme_bw() + facet_wrap(~Location,scales = "free_y")

ggplot(f.MAG.mdf2,aes(x=ratio,fill=MMRatioGroup)) + 
  geom_histogram() + theme_bw() + facet_wrap(~Location.Notes,scales="free") +
  theme(text=element_text(family="STHeiti"))
```


## PCA
```{r}

funBac.prf <- rbind(as.data.frame(p.prf[,-ncol(p.prf)]),as.data.frame(f.MAG.prf[1,-1,F]))

funbac.gdf <- t(funBac.prf)
funbac.gdf <- funbac.gdf[,which(colSums(funbac.gdf)>20)]
funbac.gdf.pam <- pam(funbac.gdf, k=4)
#plot(funbac.gdf.pam)

#funBac.prf$taxid <- rownames(funBac.prf)

pca2 <- autoplot(funbac.gdf.pam,data=as.data.frame(funbac.gdf),alpha=.3,
               loadings=T,loadings.label=T,frame=T,frame.type="norm",
               loadings.colour="black") + theme_bw()
pca2
```
```{r}
funbac.gdf.pam3 <- pam(funbac.gdf, k=3)
#plot(funbac.gdf.pam)

#funBac.prf$taxid <- rownames(funBac.prf)

pca3 <- autoplot(funbac.gdf.pam3,data=as.data.frame(funbac.gdf),alpha=.3,
               loadings=T,loadings.label=T,frame=T,frame.type="norm",
               loadings.colour="black") + theme_bw()
pca3
```

### PCA loading
```{r}
loadings <- layer_data(pca2,3)
loadings$v <- loadings$x^2+loadings$y^2
loading <- loadings[rev(order(loadings$v)),]
head(loading,11)
```
### top11 taxons

Additional fungi:
d__Eukaryota; d1__Opisthokonta; k__Fungi; Dikarya; k1__Ascomycota; p__saccharomyceta; p1__Pezizomycotina; p2__leotiomyceta; p3__dothideomyceta; c__Dothideomycetes; c1__Dothideomycetidae; o__Myriangiales; f__Myriangiaceae; g__Myriangium;

```{r}
top11.taxon <- loading[1:11,c("label","x","y")]
top11.taxon$domain <- c(rep("Archaea",3),"Eukaryota",rep("Archaea",3),
                        rep("Bacteria",3),"Archaea")
top11.taxon$class <- c("Nitrososphaeria","Nanoarchaeia","Nanoarchaeia","Dothideomycetes",
                       rep("Nitrososphaeria",3),"Gammaproteobacteria",
                       "Alphaproteobacteria","Zixibacteria;MSB-5A5","Nitrososphaeria")
top11.taxon$contributes <- seq(1,11)
top11.taxon$taxon <- factor(
  paste0("top",top11.taxon$contributes,":",top11.taxon$class,":",top11.taxon$MAGID),
  levels=paste0("top",top11.taxon$contributes,":",top11.taxon$class,":",top11.taxon$MAGID))


top11.prf.mdf <- melt(
  as.matrix(funBac.prf[which(rownames(funBac.prf)%in%top11.taxon$label),]))
top11.prf.mdf$value <- as.numeric(top11.prf.mdf$value)
pca.11.cluster <- as.data.frame(funbac.gdf.pam$clustering)
pca.11.cluster$fmtID <- rownames(pca.11.cluster)
colnames(pca.11.cluster)[1] <- "PCA.cluster"

colnames(top11.prf.mdf) <- c("MAGID","fmtID","FPKM")

top11.mdf <- merge(
  merge(top11.prf.mdf,top11.taxon,by.x="MAGID",by.y="label"),
  merge(f.MAG.mdf2,pca.cluster,by="fmtID"),by="fmtID") 
```


```{r,warning=F}
ggplot(top11.mdf,aes(x=FPKM)) + geom_density(aes(color=factor(PCA.cluster))) +
  facet_wrap(~taxon,scales="free") + scale_x_log10() + theme_bw()
```
```{r,warning=F}
ggplot(top11.mdf,aes(x=Location.Notes,y=FPKM)) + geom_boxplot(aes(fill=Location.Notes)) +
  facet_wrap(~taxon,scales="free") + scale_y_log10() + theme_bw() +
  theme(text=element_text(family="STHeiti"),axis.text.x = element_text(angle = -90,hjust=0))
```
```{r,warning=F}
top11.mdf$loc3D <- paste(top11.mdf$Longitude,top11.mdf$Latitude,top11.mdf$Depth,sep=".")
ggplot(top11.mdf%>%filter(grepl("南坡",Location.Notes)&No.dive=="FDZ066"),
       aes(x=interaction(pushcore,loc3D),y=FPKM)) +
  geom_boxplot(aes(fill=interaction(pushcore,loc3D))) +
  geom_jitter() + stat_compare_means() +
  facet_wrap(~taxon,scales="free") + scale_y_log10() + theme_bw() +
  theme(text=element_text(family="STHeiti"),axis.text.x = element_blank())

ggplot(top11.mdf%>%filter(grepl("南坡",Location.Notes)&No.dive=="FDZ069"),
       aes(x=interaction(pushcore,loc3D),y=FPKM)) +
  geom_boxplot(aes(fill=interaction(pushcore,loc3D))) +
  geom_jitter() + stat_compare_means() +
  facet_wrap(~taxon,scales="free") + scale_y_log10() + theme_bw() +
  theme(text=element_text(family="STHeiti"),axis.text.x = element_blank())

```
### Cor with bac
```{r}
bac.metadata.df <- top10.mdf%>%filter(grepl("top1:",taxon))
bac.metadata.df <- bac.metadata.df[,c(1,25:28)]

f.MAG.mdf3 <- merge(bac.metadata.df,f.MAG.mdf2,by="fmtID")
```

```{r}
funBac.top.prf <- funBac.prf[which(rownames(funBac.prf)%in%loading$label[1:11]),]
funBac.top.cor <- cor(t(funBac.top.prf),method="s")

pheatmap(funBac.top.cor,display_numbers = T)
```
```{r}
library(ggpmisc)
ggplot(f.MAG.mdf3,aes(x=TheFungi.MAG,y=MAG.shannon)) + 
  geom_point(aes(color=MMRatioGroup)) + facet_wrap(~Location) +
  stat_smooth(method = "lm", se = TRUE) +
  stat_cor(label.y=0,method="spearman") +
  theme_bw() + 
  xlab("FPKM of The Fungi") +
  ylab("prokaryotic microbiome shannon index")

ggplot(f.MAG.mdf3,aes(x=TheFungi.MAG,y=MAG.shannon)) + 
  geom_point(aes(color=MMRatioGroup)) + facet_wrap(~Location) +
  scale_x_log10() + theme_bw() + 
  xlab("FPKM of The Fungi") +
  ylab("prokaryotic microbiome shannon index")
```

## PERMANOVA


```{r}
select.vars <- c("fmtID","Depth","Longitude","Latitude","Collector",
                 "No.dive","Collect_loc","LayerDepth","LayerHeigth","day",
                 "TheFungi.MAG","TheFungi.Mitochondrion","ratio")
select.metadata.df <- f.MAG.mdf3[,select.vars]
rownames(select.metadata.df) <- select.metadata.df$fmtID
select.metadata.df <- select.metadata.df%>%filter(!is.na(Depth))

select.samples <- intersect(
  select.metadata.df$fmtID,
  intersect(colnames(s2.prf),f.MAG.mdf3$fmtID))

select.metadata.df <- select.metadata.df[select.samples,]
select.metadata.df$Collect_loc[which(is.na(select.metadata.df$Collect_loc))] <- "Unk"
select.metadata.df$LayerHeigth[which(is.na(select.metadata.df$LayerHeigth))] <- 0

select.fpkm.prf <- s2.prf[,select.samples]

select2.bymargin.permanova <- adonis2(
  t(select.fpkm.prf[s2.contigs[1:4],]) ~ Longitude, data = select.metadata.df[,1:3])
```

```{r}
select.vars <- c("fmtID","Depth","Longitude","Latitude","Collector",
                 "No.dive","Collect_loc","LayerDepth","LayerHeigth","day",
                 "TheFungi.MAG","TheFungi.Mitochondrion","ratio")
select.metadata.df <- f.MAG.mdf3[,select.vars]
rownames(select.metadata.df) <- select.metadata.df$fmtID
select.metadata.df <- select.metadata.df%>%filter(!is.na(Depth))

select.samples <- intersect(
  select.metadata.df$fmtID,
  intersect(colnames(p.prf),f.MAG.mdf3$fmtID))

select.metadata.df <- select.metadata.df[select.samples,]
select.metadata.df$Collect_loc[which(is.na(select.metadata.df$Collect_loc))] <- "Unk"
select.metadata.df$LayerHeigth[which(is.na(select.metadata.df$LayerHeigth))] <- 0
select.fpkm.prf <- p.prf[,select.samples]

select.permanova.RData.file <- "./select.byterm.permanova.RData"
if(file.exists(select.permanova.RData.file)){
  load(select.permanova.RData.file)
}else{
  library(vegan)
  
  select.byterm.permanova <- adonis2(t(select.fpkm.prf) ~ Depth+Longitude+Latitude+Collector+No.dive+Collect_loc+LayerDepth+LayerHeigth+TheFungi.MAG+TheFungi.Mitochondrion+ratio, 
                              by="term",
                              data = select.metadata.df)
  
  select.bymargin.permanova <- adonis2(t(select.fpkm.prf) ~ Depth+Longitude+Latitude+Collector+No.dive+Collect_loc+LayerDepth+LayerHeigth+TheFungi.MAG+TheFungi.Mitochondrion+ratio, 
                              by="margin",
                              data = select.metadata.df)
  save(select.byterm.permanova,select.bymargin.permanova,
       file = select.permanova.RData.file)
}
```



# 3.Circular DNA
Choose represent contigs:  
FDZ070-GBkY16-18.k141_555386
FDZ072-WuWY8-10.k141_195567
FDZ070-GBkY16-18.bin.38.fa      k141_555386     31556
FDZ070-GBkY16-18.bin.38.fa      k141_771931     47277
FDZ070-GBkY16-18.bin.38.fa      k141_49412      1085

```{bash,eval=FALSE}
bgzip -dc profiles/MEER1225.fpkm.profile.gz|grep -E "OTU|FDZ070-GBkY16-18|FDZ072-WuWY8-10" > profiles/MEER1225.select2.fpkm.prof
```

```{r}
s2.prf <- read.table("profiles/MEER1225.select2.fpkm.prof",
                    header=T,row.names = 1)
s2.df <- as.data.frame(t(s2.prf))

s2.contigs <- c("FDZ070-GBkY16-18.38.k141_555386",
                "FDZ070-GBkY16-18.38.k141_771931",
                "FDZ070-GBkY16-18.38.k141_49412",
                "FDZ072-WuWY8-10.14.k141_195567")
s2.df <- s2.df[,which(colnames(s2.df)%in%s2.contigs)]
s2.df$sample <- rownames(s2.df)

```

```{r}
p1 <- ggplot(s2.df,aes(x=`FDZ070-GBkY16-18.38.k141_555386`,
                 y=`FDZ072-WuWY8-10.14.k141_195567`)) + 
  geom_point() + stat_cor()

p2 <- ggplot(s2.df,aes(x=`FDZ070-GBkY16-18.38.k141_555386`,
                 y=`FDZ070-GBkY16-18.38.k141_771931`)) + 
  geom_point() + stat_cor()

p3 <- ggplot(s2.df,aes(x=`FDZ070-GBkY16-18.38.k141_555386`,
                 y=`FDZ070-GBkY16-18.38.k141_49412`)) + 
  geom_point() + stat_cor()

p4 <- ggplot(s2.df,aes(x=`FDZ070-GBkY16-18.38.k141_771931`,
                 y=`FDZ070-GBkY16-18.38.k141_49412`)) + 
  geom_point() + stat_cor()

cowplot::plot_grid(p1,p2,p3,p4)
```


## Cor with phe
```{r}
s2.MAG.mdf <- merge(f.MAG.mdf3,s2.df,by.x="fmtID",by.y="sample")

the.cors <- cor(
  s2.MAG.mdf[,c(2:5,7:9,17,18,20,22:24,26:29)],
  method="s",use="complete.obs")
pheatmap(the.cors,cluster_cols = F,cluster_rows = F)
```
```{r}
funBac.prf
```

#Fin.



