---
title: "mwas test"
author: "Ciao"
output: html_notebook
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(ggplot2)
library(ggpubr)
library(ggbiplot) # devtools::install_github("vqv/ggbiplot")
library(cowplot)
library(reshape2)
library(RColorBrewer)
library(plyr)
library(dplyr)
library(vegan)
library(ape)
library(pheatmap)
library(readxl)
if (!require(devtools)) install.packages("devtools")
if (!require(ggvenn)) devtools::install_github("yanlinlin82/ggvenn")
library(ggvenn)
library(plotly)



```

# Metadata

```{r}
metadata.df <- read_excel("./TS21-merged-metadata.xlsx","subset",
    col_types=c(
      "numeric","text","numeric","numeric","numeric",
      "text","text","date","text","text","numeric","numeric",
      "text","numeric","numeric","numeric","numeric","numeric","numeric"),
    na="NA")

metadata.df$fmtID <- gsub("-",".",metadata.df$fmtID,fixed = T)
metadata.df$fmtID <- gsub("[()]","_",metadata.df$fmtID,perl=T)
metadata.df$Depth <- - metadata.df$Depth
metadata.df$Collect_date <- as.POSIXlt(metadata.df$Collect_date,
                                       formats = "%Y-%m-%d")

metadata.df$day <- as.numeric(difftime(metadata.df$Collect_date,"2021-08-21 8:0:0",units="days"))

summary(metadata.df)
```

## No.dive
```{r}
fig1 <- plot_ly(metadata.df, x=~Longitude, y=~Latitude, z= ~ Depth, color= ~No.dive)

fig1
```
```{r}
repr.dive.df <- ddply(
  unique(metadata.df[,c("No.dive","Location","Longitude","Latitude")]),
  c("No.dive","Location"),summarize,
  Longitude=mean(Longitude), Latitude=mean(Latitude))

#ggplot(repr.dive.df%>%filter(grepl("Mar",Location)),
ggplot(repr.dive.df,
       aes(x=Longitude,y=Latitude)) + 
  geom_point(aes(color=No.dive)) +
  geom_text(aes(label=No.dive),hjust=1.1) + theme_bw()

ggplot(repr.dive.df%>%filter(grepl("Mar",Location)),
       aes(x=Longitude,y=Latitude)) + 
  geom_point(aes(color=No.dive)) +
  geom_text(aes(label=No.dive),hjust=1.1) + theme_bw()
```


## Collector
```{r}
fig2 <- plot_ly(metadata.df, x=~Longitude, y=~Latitude, z= ~ Depth, color= ~Collector)

fig2
```

## Collect_loc
```{r}
metadata.df$Collect_loc.s <- substr(metadata.df$Collect_loc,1,1)
fig4 <- plot_ly(metadata.df, x=~Longitude, y=~Latitude, z= ~ Depth, 
                color= ~Collect_loc.s)

fig4
```

# MAG fpkm profile
```{r}
p.prf <- read.table(gzfile("profiles/MEER1225.d95.fpkm.MAG.profile.gz"),
                    header=T,row.names = 1)

p.shannon <- data.frame(
  MAG.shannon=diversity(p.prf,"shannon",MARGIN=2),
  MAG.simpson=diversity(p.prf,"simpson",MARGIN=2),
  MAG.invsimpson=diversity(p.prf,"invsimpson",MARGIN=2)
)
p.shannon$fmtID <- rownames(p.shannon)

mdf <-merge(metadata.df,p.shannon,by="fmtID")
```

## Alpha diversity
```{r}
ggplot(mdf,aes(x=MAG.shannon,color=Location)) + geom_density() + theme_bw()
ggplot(mdf,aes(x=MAG.simpson,color=Location)) + geom_density() + theme_bw()
ggplot(mdf,aes(x=MAG.invsimpson,color=Location)) + geom_density() + theme_bw()
```
总体分布上来看，大部分样本的shannon index可以达到4以上。马沟西部整体丰富度最高，马沟东部次之。
菲律宾海盆样本的多样性相对较低。

```{r}
ggplot(mdf,aes(x=MAG.shannon,y=Depth,shape=Location,color=No.dive)) + geom_point()
ggplot(mdf,aes(x=MAG.simpson,y=Depth,shape=Location,color=No.dive)) + geom_point()
ggplot(mdf,aes(x=MAG.invsimpson,y=Depth,shape=Location,color=No.dive)) + geom_point()
```
Alpha diversity 与下潜深度的关系。部分低多样性样本点在各个采样区域(Location)都有发现。


```{r}
ggplot(mdf,aes(x=MAG.shannon,y=LayerDepth,color=Collector)) + stat_density_2d() + geom_point(aes(shape=Location))
ggplot(mdf,aes(x=MAG.simpson,y=LayerDepth,color=Collector)) + stat_density_2d() + geom_point(aes(shape=Location))
ggplot(mdf,aes(x=MAG.invsimpson,y=LayerDepth,color=Collector)) + stat_density_2d() + geom_point(aes(shape=Location))
```

Alpha diversity 与下潜深度的关系。部分低多样性样本点可能与特定采样人员参与的特定潜次有关。
- 低多样性样本点多来自菲律宾海盆样本；
- 低多样性样本点多集中在10cm以下的pushcore分层中。

```{r}
ggplot(mdf,aes(x=Collector,y=MAG.shannon,fill=Collector)) + geom_violin(alpha=.5) + geom_boxplot(width=.1) + 
  theme_bw() + facet_grid(.~Location,space="free",scale="free")
```

Alpha diversity 在采样人员分组之间的比较。低多样性样本点呈现两种情况：  
- 在菲律宾海盆地区的低样本点未超出整体样本分布区间(95% CI)，具有一定普遍性；
- 在马沟内，样本整体多样性高；低丰度样本多为离群点(超出95% CI)，可能是异常状态。

* 95% CI: value < 1.58 * IQR / sqrt(n)

```{r,warning=FALSE}
ggplot(mdf,aes(x=Collector,y=MAG.shannon,fill=Collector)) + geom_violin() + geom_jitter() +
  theme_bw() + facet_wrap(~Location+No.dive,ncol=8,scale="free_x")
```

Alpha diversity 在潜次之间的比较。低多样性样本集中出现于特定潜次。其中FDZ072（Z.WS） & FDZ076(Z.WJ)出现频率最高。


```{r}
ggplot(mdf%>%filter(No.dive%in%c("FDZ072","FDZ069","FDZ076")),
       aes(x=interaction(Depth,Characteristics),y=LayerDepth,color=MAG.shannon)) + 
  geom_point(aes(shape=No.dive)) +
  scale_color_distiller(palette = "RdBu") + facet_grid(.~No.dive,scale="free",space="free") +
  theme_bw() + theme(axis.text.x = element_text(angle=-90,hjust=0))
```

低多样性样本集中出现于特定pushcore的深层位置中。



## PCA

```{r}
library(cluster)

gdf <- t(p.prf)
gdf1 <- gdf[,which(colSums(gdf)>20)]
g.pam <- pam(gdf1, k=3)
plot(g.pam)
```


```{r}
library(ggfortify)

p.prf1 <- as.data.frame(p.prf)
p.prf$taxid <- rownames(p.prf1)

p1 <- autoplot(g.pam,data=as.data.frame(gdf),alpha=.3,
               loadings=T,loadings.label=T,frame=T,frame.type="norm",
               loadings.colour="black")
p1
```



```{r}
loadings <- layer_data(p1,3)
loadings$v <- loadings$x^2+loadings$y^2
loading <- loadings[rev(order(loadings$v)),]
head(loading,10)
```


## taxonoming
```{r}
binmap.tsv <- read.table("drep_all/drep_all_95.binmap.tsv",
                         col.names = c("bin","cluster"))
drep_all_95.Cdb.csv <- read.csv("drep_all/drep_all_95.Cdb.csv")
drep_all_95.Cdb.csv$bin <- gsub(".fa","",gsub(".bin","",drep_all_95.Cdb.csv$genome))
```

```{r}
gtdbtk.ar53 <- read.table(gzfile("drep_all/drep_all_99.gtdbtk.ar53.summary.tsv.gz"),
                            sep="\t",header = T)
gtdbtk.ar53$bin <- gsub(".bin","",gtdbtk.ar53$user_genome)

bin.gtdbtk.ar53 <- merge(drep_all_95.Cdb.csv[,c("secondary_cluster","bin")],
                           gtdbtk.ar53[,c("user_genome","classification","bin")],by='bin')

bin.gtdbtk.ar53%>%filter(secondary_cluster%in%loading$label[1:6])
```

按loading向量长度排序，前6名、第10名MAG均为古菌。

注释信息简单汇总如下：
2455_2:   d__Archaea;p__Thermoproteota;c__Nitrososphaeria;o__Nitrososphaerales;
          f__Nitrosopumilaceae;g__DRGT01;s__DRGT01 sp016838795
2626_4:   d__Archaea;p__Nanoarchaeota;c__Nanoarchaeia;o__Pacearchaeales;
          f__GW2011-AR1;g__PSPM01;s__
2537_0:   d__Archaea;p__Nanoarchaeota;c__Nanoarchaeia;o__SCGC-AAA011-G17;
          f__JACPIM01;g__;s__
2466_1:   d__Archaea;p__Thermoproteota;c__Nitrososphaeria;o__Nitrososphaerales;
          f__Nitrosopumilaceae;g__Nitrosopumilus;s__Nitrosopumilus sp013390905
2455_1:   d__Archaea;p__Thermoproteota;c__Nitrososphaeria;o__Nitrososphaerales;
          f__Nitrosopumilaceae;g__DRGT01;s__
2467_1:   d__Archaea;p__Thermoproteota;c__Nitrososphaeria;o__Nitrososphaerales;
          f__Nitrosopumilaceae;
2465_1:   d__Archaea;p__Thermoproteota;c__Nitrososphaeria;o__Nitrososphaerales;
          f__Nitrosopumilaceae;g__Nitrosopumilus;s__

其中主要成员来自*氨氧化古菌*Nitrososphaeria、*亚硝化侏儒菌科*(Nitrosopumilaceae)和*纳古菌门*（Nanoarchaeota）。

综上，PCA聚类主要分为三种类型：
- c1, 由氨氧化古菌（2455_2）主导,具体为DRGT01 sp016838795
- c2, 由纳古菌（2626_4，2537_0）主导，但对应科不同，具体种未知。
- c3, 由氨氧化古菌（2455_2）主导,具体为Nitrosopumilus sp013390905


```{r}
gtdbtk.bac120 <- read.table(gzfile("drep_all/drep_all_99.gtdbtk.bac120.summary.tsv.gz"),
                            sep="\t",header = T)
gtdbtk.bac120$bin <- gsub(".bin","",gtdbtk.bac120$user_genome)

bin.gtdbtk.bac120 <- merge(drep_all_95.Cdb.csv[,c("secondary_cluster","bin")],
                           gtdbtk.bac120[,c("user_genome","classification","bin")],by='bin')

bin.gtdbtk.bac120%>%filter(secondary_cluster%in%loading$label[1:6])
bin.gtdbtk.bac120%>%filter(secondary_cluster%in%loading$label[7:9])
```
191_1:    d__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__UBA4486;f__UBA4486;g__SMWN01;s__SMWN01 sp004356415
1471_1:   d__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__;f__;g__;s__
2340_0:   d__Bacteria;p__Zixibacteria;c__MSB-5A5;o__GN15;f__FEB-12;g__B120-G9;s__

191_1:    变形菌门；γ-变形菌纲
1471_1:   变形菌门；α-变形菌纲
2340_0:   河床菌门；

### Top10 distribution
```{r}
top10.taxon <- loading[1:10,c("label","x","y")]
top10.taxon$domain <- c(rep("Archaea",6),rep("Bacteria",3),"Archaea")
top10.taxon$class <- c("Nitrososphaeria","Nanoarchaeia","Nanoarchaeia",
                       rep("Nitrososphaeria",3),"Gammaproteobacteria",
                       "Alphaproteobacteria","Zixibacteria;MSB-5A5","Nitrososphaeria")
top10.taxon$contributes <- seq(1,10)
top10.taxon$taxon <- factor(paste0("top",top10.taxon$contributes,":",top10.taxon$class,":",top10.taxon$MAGID),
                            levels=paste0("top",top10.taxon$contributes,":",top10.taxon$class,":",top10.taxon$MAGID))


top10.prf.mdf <- melt(as.matrix(p.prf[which(rownames(p.prf)%in%top10.taxon$label),]))
pca.cluster <- as.data.frame(g.pam$clustering)
pca.cluster$fmtID <- rownames(pca.cluster)
colnames(pca.cluster)[1] <- "PCA.cluster"

colnames(top10.prf.mdf) <- c("MAGID","fmtID","FPKM")

top10.mdf <- merge(
  merge(top10.prf.mdf,top10.taxon,by.x="MAGID",by.y="label"),
  merge(mdf,pca.cluster,by="fmtID"),by="fmtID")
```

```{r}
ggplot(top10.mdf,aes(x=FPKM,color=taxon)) + 
  geom_density() + scale_x_log10()
```
```{r}
ggplot(top10.mdf,aes(x=PCA.cluster,y=FPKM,fill=factor(PCA.cluster))) + 
  geom_violin() + geom_boxplot(width=.2) + scale_y_log10() + facet_wrap(.~taxon,scale="free",ncol=5) +
  theme_bw() + scale_fill_brewer(palette="Set1") +
  stat_compare_means(label = "p.signif")
```

```{r}
table(top10.mdf$PCA.cluster,top10.mdf$Location)/10

ggplot(top10.mdf,aes(x=PCA.cluster,y=..count../10)) + geom_bar(aes(fill=factor(PCA.cluster))) + 
  facet_grid(.~Location) + theme_bw() + scale_fill_brewer(palette="Set1")
```

聚类分布有以下特征：
- 菲律宾海盆南部几乎以C1为主；
- 菲律宾海盆南北部以C1为主，少量C2，几乎无C3；
- 马沟东部以C1为主，C3次之，极少C2；
- 马沟西部几乎均为C3类型

此外，C3在马沟区域比例明显高于菲律宾海盆地区，提示C3可能是马沟特有的群落结构。
马沟西部完全没有观察到C2类型的样本，可能跟该地区的生境有所关联。


```{r}
ggplot(top10.mdf%>%filter(contributes<5),aes(x=contributes,y=FPKM,fill=taxon)) + geom_boxplot() + 
  facet_grid(.~Location) + theme_bw() + scale_y_log10() + stat_compare_means(label = "p.signif")
```




#Fin.
# ##########

# Kraken2 profile


Check `README.md` for more details how I careated the following profiles.

Load raw count profile
```{r}
#index
refidx <- read.table("./drep_all/drep_all_95.rename.fasta.gz.fai",
                     col.names = c("refID","length","chrPt","lineLen","LineLen2"))

# raw bin count
raw.bin.prof <- as.matrix(read.table(
  gzfile("profiles/select11dives.d95.sum.bin.raw.count.gz")))
samples <- colnames(raw.bin.prof)
bins <- rownames(raw.bin.prof)
```

Get refid's bin ID:
```{r}
parse.one <- function(res, result) {
  m <- do.call(rbind, lapply(seq_along(res), function(i) {
    if(result[i] == -1) return("")
    st <- result[i]
    substring(res[i], st, st + attr(result, "match.length")[i] - 1)
  }))
  return(m)
}

refidx$bin <- as.character(parse.one(refidx$refID, 
  regexpr("^.+\\.\\d+", refidx$refID, perl = TRUE)))
```
```{r}

ggplot(refidx,aes(x=length)) + geom_density() + scale_x_log10()
```

filter metadata:
```{r}
select.metadata.df <- metadata.df%>%filter(fmtID%in%samples)
select.samples <- select.metadata.df$fmtID
select.bin.raw.prof <- 
  raw.bin.prof[,colnames(raw.bin.prof)%in%select.samples]
ggplot(select.metadata.df,aes(x=Location,y=Depth)) + geom_boxplot()

```
## Show location
Specific set north/south slope sampling site of Mariana trench.

```{r}
select.metadata.df$slope <- ifelse(
  select.metadata.df$No.dive=="FDZ064","NorthSlope",
  ifelse(
    select.metadata.df$No.dive%in%c("FDZ063","FDZ061"),"ravineBotton",
    ifelse(
      select.metadata.df$No.dive=="FDZ062","SouthSlope",NA
    )
  )
)
```


```{r}
repr.dive.df <- unique(select.metadata.df[,c("No.dive","Location","Longitude","Latitude")])

repr.dive.df <- ddply(repr.dive.df,c("No.dive","Location"),summarize,
                      Longitude=median(Longitude),Latitude=median(Latitude))

ggplot(repr.dive.df,aes(x=Longitude,y=Latitude)) + 
  geom_point(aes(color=No.dive)) +
  geom_text(aes(label=No.dive),hjust=1) + theme_classic()

if(do.write){
  ggsave("figures/select.dives.map.pdf",width = 10,height=6)
}
ggplot(repr.dive.df%>%filter(grepl("Mar",Location)),aes(x=Longitude,y=Latitude)) + 
  geom_point(aes(color=No.dive)) +
  geom_text(aes(label=No.dive),hjust=1)
```

## Unmapping rate
```{r}
unmappedReads <- data.frame(
  fmtID       = colnames(select.bin.raw.prof),
  unmap.count = select.bin.raw.prof["UNMAPPED",],
  all.count   = colSums(select.bin.raw.prof),
  unmap.rate  = select.bin.raw.prof["UNMAPPED",]/colSums(select.bin.raw.prof))

unmappedReads$mappingRate <- 1 - unmappedReads$unmap.rate

unmap.merge.df <- merge(select.metadata.df,unmappedReads,by="fmtID")
```

```{r}
ggplot(unmap.merge.df,aes(x=No.dive,y=mappingRate)) + 
  geom_boxplot(aes(fill=No.dive)) + 
  facet_grid(.~Location,scales = "free",space = "free") + ylim(c(0,1)) +
  theme_bw() + theme(axis.text.x = element_text(angle=-90,hjust=0)) 
```

```{r}
ggplot(unmap.merge.df%>%filter(!is.na(slope)),
       aes(x=slope,y=mappingRate,fill=No.dive)) + geom_boxplot()
```


## Normalize profile
```{r}
library(dplyr)
refBin <- ddply(refidx[,c("bin","length")],"bin",summarise,
                seqs=length(bin),length=sum(length))

select.bin.norm.prof <- select.bin.raw.prof[-nrow(select.bin.raw.prof),]
select.bin.norm.prof <-
  select.bin.norm.prof[order(rownames(select.bin.norm.prof)),]

# Make sure the bin name orders are consistent
length(which(rownames(select.bin.norm.prof)==refBin$bin)) ==nrow(select.bin.norm.prof)

tmp <- 
  apply(select.bin.norm.prof,2,function(x){x*150/refBin$length})
select.bin.norm.prof <- 
  apply(tmp,2,function(x){x/sum(x)})
```

```{r}
p.0 <- ggplot(refBin,aes(x=seqs,y=length)) + geom_density2d() +
  scale_x_continuous(breaks=seq(0,6)*100,limits=c(0,600)) + 
  scale_y_continuous(breaks=seq(0,6)*1e6,limits=c(1e5,6e6)) + theme_bw()

p.x <- ggplot(refBin,aes(x=seqs)) + geom_density() +
  scale_x_continuous(breaks=seq(0,6)*100,limits=c(0,600)) + 
  scale_y_reverse() + theme_bw()

p.y <- ggplot(refBin,aes(x=..density..,y=length)) + geom_density() +
  scale_y_continuous(breaks=seq(0,6)*1e6,limits=c(1e5,6e6)) + theme_bw()

cowplot::plot_grid(p.0,p.y,p.x,rel_widths=c(3,1),rel_heights=c(3,1))
```

### alpha diversity
```{r}
library(vegan)

select.bin.alpha <- data.frame(
  fmtID=colnames(select.bin.norm.prof),
  shannon = diversity(select.bin.norm.prof,"shannon",MARGIN = 2),
  simpson = diversity(select.bin.norm.prof,"simpson",MARGIN = 2),
  invsimpson = diversity(select.bin.norm.prof,"invsimpson",MARGIN = 2)
  )

select.bin.alpha.melt <- melt(select.bin.alpha,id.vars = "fmtID",
                              variable.name = "method",value.name = "diversity")
bin.alpha.merge.df <- merge(select.metadata.df,select.bin.alpha.melt,by="fmtID")

```
```{r}
ggplot(bin.alpha.merge.df,aes(x=No.dive,y=diversity)) + 
  geom_boxplot(aes(fill=No.dive)) + 
  facet_grid(method~Location,scales = "free",space = "free_x") +
  theme_bw() + theme(axis.text.x = element_text(angle=-90,hjust=0))
```

```{r}
ggplot(bin.alpha.merge.df%>%filter(No.dive%in%c("FDZ061","FDZ063")),
       aes(x=No.dive,y=diversity)) + 
  geom_boxplot(aes(fill=No.dive)) + 
  facet_grid(method~Location,scales = "free",space = "free_x") +
  theme_bw() + theme(axis.text.x = element_text(angle=-90,hjust=0)) + 
  stat_compare_means()
```

```{r}
bin.alpha.merge.df <- bin.alpha.merge.df[order(bin.alpha.merge.df$LayerDepth),]
ggplot(bin.alpha.merge.df,aes(x=LayerDepth,y=diversity)) + 
  geom_path(aes(group=interaction(No.dive,Characteristics)),alpha=.7) +
  geom_point(aes(color=interaction(No.dive,Characteristics))) +
  facet_grid(method~Location,scales = "free",space = "free_x") +
  theme_bw()
```

```{r}
ggplot(bin.alpha.merge.df%>%filter(method=="shannon")) + 
  geom_bar(stat="identity",aes(x=LayerDepth,y=diversity,fill=Location),width=2) +
  facet_wrap(~Location+interaction(No.dive,Characteristics)) +
  ylim(c(0,10)) + theme_bw()
```

## correlation
```{r}
library(pheatmap)
bins.cor <- cor(t(select.bin.norm.prof))
sams.cor <- cor(select.bin.norm.prof,method="spearman")

sams.anno <- as.data.frame(select.metadata.df)
rownames(sams.anno) <- sams.anno$fmtID
sams.anno.color <- sams.anno[,c("Red","Green","Blue","Yellow","White","Black")]

pheatmap(sams.cor,annotation_col =
           sams.anno[,c("Depth","LayerDepth","Location"),F],
         show_rownames = F, show_colnames = F)
```

# Kraken2 taxonomic identification
```{r}
kk2.all.benchmark.tsv <- read.table(
  "taxon/all_passed_bins.kraken2.benchmark.tsv",
  col.names = c("bin","seqs","length","S.p","S.c","G.p","G.c","F.p","F.c",
                "O.p","O.c","C.p","P.c","P.p","P.c","K.p","K.c","D.p","D.c","R.p","R.c")
  )

kk2.all.S.benchmark.tsv <- read.table(
  "taxon/all_passed_bins.kraken2.S.benchmark.tsv",
  col.names = c("bin","seqs","length","taxonID","TP","FP","FN","purify","completeness")
  )

bin.d99.S.benchmark.tsv <- read.table(
  "taxon/drep_99.cluster.S.benchmark.tsv",
  col.names = c("bin","seqs","length","taxonID","TP","FP","FN","purify","completeness")
  )
gtdbtk.d99.S.benchmark.tsv <- read.table(
  "taxon/drep_99.gtdbtk.S.benchmark.tsv",
  col.names = c("bin","seqs","length","taxonID","TP","FP","FN","purify","completeness")
  )

kk2.d95.S.benchmark.tsv <- read.table(
  "taxon/drep_95.kraken2.S.benchmark.tsv",
  col.names = c("bin","seqs","length","taxonID","TP","FP","FN","purify","completeness")
  )

kk2.all.classify.df <- read.table(
  gzfile("taxon/all_passed_bins.kraken2.detail.gz"),
  col.names = c("Classify","ID","RankID","length",
                "Rank_S","Rank_G","Rank_F","Rank_O","Rank_C","Rank_P","Rank_K","Rank_D","Rank_R")
  )

```

```{r}
merge.df <- rbind(
  data.frame(group="kk2.all",Rank="S",kk2.all.S.benchmark.tsv),
  data.frame(group="cluster.d99",Rank="S",bin.d99.S.benchmark.tsv),
  data.frame(group="gtdbtk.d99",Rank="S",gtdbtk.d99.S.benchmark.tsv),
  data.frame(group="kk2.d95",Rank="S",kk2.d95.S.benchmark.tsv)
)
mean(kk2.all.S.benchmark.tsv$purify)
mean(kk2.all.S.benchmark.tsv$completeness)

length(which(kk2.all.S.benchmark.tsv$completeness>=0.5&kk2.all.S.benchmark.tsv$purify>0.9))

ggplot(merge.df,aes(x=group,y=purify,fill=group)) + geom_boxplot() + theme_bw() + coord_flip() + 
  scale_y_continuous(breaks=c(0,0.5,1),labels = c(0,50,100)) + xlab("Purity(%)")

ggplot(merge.df,aes(x=group,y=completeness,fill=group)) + geom_boxplot() + theme_bw() + coord_flip() + 
  scale_y_continuous(breaks=c(0,0.5,1),labels = c(0,50,100))
```
```{r}
mean(kk2.d95.S.benchmark.tsv$purify)
mean(kk2.d95.S.benchmark.tsv$completeness)

length(which(kk2.d95.S.benchmark.tsv$completeness>=0.5&kk2.d95.S.benchmark.tsv$purify>0.9))

```

```{r}
ggplot(kk2.all.S.benchmark.tsv,aes(x=purify,y=completeness)) + 
  stat_density2d(geom="raster",contour = F,h=c(.5,.5))

```
```{r}
mean(gtdbtk.d99.S.benchmark.tsv$purify)
mean(gtdbtk.d99.S.benchmark.tsv$completeness)

length(which(gtdbtk.d99.S.benchmark.tsv$completeness>=0.5&gtdbtk.d99.S.benchmark.tsv$purify>0.9))

```

```{r}
ggplot(kk2.d95.S.benchmark.tsv,aes(x=purify)) + geom_boxplot()
ggplot(kk2.d95.S.benchmark.tsv,aes(x=completeness)) + geom_boxplot()

```



### Curate
```{r}
kk2vamb.df <- merge()
```


